script/bump-version.js

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env node
+
+const { GitProcess } = require('dugite')
+const utils = require('./lib/version-utils')
+const plist = require('plist')
+const fs = require('fs')
+const semver = require('semver')
+const path = require('path')
+const { promisify } = require('util')
+const minimist = require('minimist')
+
+const writeFile = promisify(fs.writeFile)
+const readFile = promisify(fs.readFile)
+
+function parseCommandLine () {
+  let help
+  const opts = minimist(process.argv.slice(2), {
+    string: [ 'bump', 'version' ],
+    boolean: [ 'dryRun', 'help' ],
+    alias: { 'version': ['v'] },
+    unknown: arg => { help = true }
+  })
+  if (help || opts.help || !opts.bump) {
+    console.log(`
+      Bump release version number. Possible arguments:\n
+        --bump=patch to increment patch version\n
+        --version={version} to set version number directly\n
+        --dryRun to print the next version without updating files
+      Note that you can use both --bump and --stable  simultaneously. 
+    `)
+    process.exit(0)
+  }
+  return opts
+}
+
+// run the script
+async function main () {
+  const opts = parseCommandLine()
+  const currentVersion = await utils.getElectronVersion()
+  const version = await nextVersion(opts.bump, currentVersion)
+  const components = semver.valid(semver.coerce(version)).split('.')
+
+  let pre
+  if (version.includes('-')) {
+    pre = `-${version.split('-')[1]}`
+    components[3] = version.split('-')[1].split('.')[1]
+  }
+
+  // print would-be new version and exit early
+  if (opts.dryRun) {
+    console.log(`new version number would be: ${version}\n`)
+    return 0
+  }
+
+  // update all version-related files
+  await Promise.all([
+    updateVersion(version),
+    updateInfoPlist(version),
+    updatePackageJSON(version),
+    tagVersion(version),
+    updateVersionH(components, pre),
+    updateWinRC(components)
+  ])
```

Comment:
Am I correct that `tagVersion` should be responsible for checking in the changes to all the other files that are being modified? If so, there's a race here — `tagVersion` should be removed from this array and ran after all the other promises are resolved.